sonar-db/src/main/java/org/sonar/db/component/ComponentValidator.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.db.component;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class ComponentValidator {
+  private static final int MAX_NAME_LENGTH = 2000;
+  private static final int MAX_KEY_LENGTH = 400;
+  private static final int MAX_QUALIFIER_LENGTH = 10;
+
+  private ComponentValidator() {
+    // prevent instantiation
+  }
+
+  public static String checkComponentName(String name) {
+    checkArgument(name.length() <= MAX_NAME_LENGTH, "Component name length (%s) is longer than the maximum authorized (%s). '%s' was provided.",
+      name.length(), MAX_NAME_LENGTH, name);
+    return name;
+  }
+
+  public static String checkComponentKey(String key) {
+    checkArgument(isComponentKeyValid(key), "Component key length (%s) is longer than the maximum authorized (%s). '%s' was provided.",
+      key.length(), MAX_KEY_LENGTH, key);
+    return key;
+  }
+
+  public static boolean isComponentKeyValid(String key) {
+    return key.length() <= MAX_KEY_LENGTH;
```

Comment:
You should use https://github.com/SonarSource/sonarqube/blob/master/sonar-core/src/main/java/org/sonar/core/component/ComponentKeys.java#L96